source/util/ilist_node.h

```diff
@@ -196,33 +199,50 @@ inline void IntrusiveNodeBase<NodeType>::RemoveFromList() {
 
 template<class NodeType>
 void IntrusiveNodeBase<NodeType>::ReplaceWith(NodeType* target) {
-  if (is_sentinel_) {
-    assert(target->next_node_ == target);
-    assert(this->is_sentinel_);
+  if (this->is_sentinel_) {
+    target->IsEmptyList();
```

Comment:
` target->IsEmptyList();`

assert(target->is_sentinel_)?

Reply:
Should be assert(target->IsEmptyList()), and that checks target->is_sentinel_.  I fixed that up.